### PR TITLE
feat: Simplify taxonomy tables

### LINKED ISSUE #346: Simplification of the `taxonomy_` tables
The following simplifications can be done easily:

* `taxonomy` can be merged into `term_taxomony` (`taxonomy` is only a reference to a `type_id`).
* `term_taxonomy` can be renamed to `taxonomy_term` or `taxonomy`
* `term` can be merged into `term_taxonomy` (`term` only stores the name)

### PR DESCRIPTION
**Pull request by kulla**
Fixes #346

## ToDo List

- [ ] Rename column `term_taxonomy_id` into `taxonomy_id` in `term_taxonomy_entity_link`
- [x] check whether frontend still needs the `taxonomyId` field, see https://serlo.github.io/unused-graphql-properties/#TaxonomyTerm.taxonomyId => I would like to delete it, since we delete the `taxonomy` table here.

### CODE REVIEW

File: src/20240608121400-simplification-of-taxonomy-tables.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+  console.log('Dropping term table')
+  await db.runSql(
+    'ALTER TABLE term_taxonomy DROP INDEX uq_term_taxonomy_unique;',
+  )
+  await db.runSql(
+    'ALTER TABLE term_taxonomy DROP FOREIGN KEY fk_term_taxonomy_term1;',
+  )
+  await db.runSql(`ALTER TABLE term_taxonomy DROP COLUMN term_id`)
+  await db.runSql(`DROP TABLE term`)
+}
```
Comment (hugotiburtino) [inline src/20240608121400-simplification-of-taxonomy-tables.ts:147]: What about renaming the table to make it slightly shorter?
```suggestion
ALTER TABLE term_taxonomy RENAME taxonomy;
}
```

Comment (kulla) [reply src/20240608121400-simplification-of-taxonomy-tables.ts:147]: We have the same idea: https://github.com/serlo/db-migrations/blob/ea8b8574a16df0b4bb9e97f7601fe03f8996deaf/src/20240608121400-simplification-of-taxonomy-tables.ts#L12

**Review by hugotiburtino — APPROVED** [review]
LGTM

File: src/20240505020500-merge-entity_revision_field-into-entity_revision.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            set content = ?, meta_title = ?, title = ?,
+                meta_description = ?, changes = ?, url = ?
+            where id = ?
+        `,
+        [
+          params['content'] ?? null,
+          params['meta_title'] ?? null,
+          params['title'] ?? null,
+          params['meta_description'] ?? params['description'] ?? null,
+          params['reasoning'] ?? params['Changes'] ?? null,
```
Comment (hugotiburtino) [inline src/20240505020500-merge-entity_revision_field-into-entity_revision.ts:55]: Any reason why we should prefer reasoning 
Some of the content of reasoning are in slate format, leading to the following result:

image

Changes are in clear text.

Comment (hugotiburtino) [reply src/20240505020500-merge-entity_revision_field-into-entity_revision.ts:55]: Kulla's words
> no particular reason, you can change it

**Review by hugotiburtino — APPROVED** [review]

**Pull request merged**